sdk/storage/azure-storage-blobs/src/blob_container_client.cpp

```diff
@@ -261,6 +261,22 @@ namespace Azure { namespace Storage { namespace Blobs {
     return response;
   }
 
+  Pageable<ListBlobsPageResult> BlobContainerClient::ListBlobs(
+      const ListBlobsOptions& options,
+      const Azure::Core::Context& context) const
+  {
+    ListBlobsPageResult pageResult;
+    pageResult.NextPageToken = options.ContinuationToken.ValueOr(std::string());
+    pageResult.m_blobContainerClient = std::make_shared<BlobContainerClient>(*this);
```

Comment:
Just suppose there are two classes A and B and they depend on each other. How are you going to define them, the only way is to

```C++
struct A {
    B* m_b;  // have to be a pointer, because B is not defined yet.
};

struct B {
    A* m_a1;  // can be a pointer or value, becasue A is already defined.
    A m_a2;
};
```

Reply:
Why does `ListBlobsPageResult` have a  `std::shared_ptr<BlobContainerClient> m_blobContainerClient;`? Why doesn't it take ownership of the client (unique ptr)?

Reply:
I don't think shared_ptr has any difference with unique_ptr in regard of ownership